### ISSUE TITLE
Add `--platform` with rosetta to pixi exec

### ISSUE THREAD

**Issue by pavelzw** [prefix-dev/pixi#2161]
### Problem description

Similar to `pixi global install`, it might make sense to have the rosetta fallback option in `pixi exec`

<img width="734" alt="image" src="https://github.com/user-attachments/assets/117acc7a-3217-4eb3-9d04-32eeb6480ca4">

https://pixi.sh/v0.30.0/reference/cli/#options_24

**Comment by AntoinePrv** [comment]
For `pixi run` would be nice too, but it would require distinguishing environments in `.pixi`.

**Comment by baszalmstra** [comment]
Pixi run should already support this. We have code that detects "the best platform" for an environment. If no osx-arm64 is available, osx-64 will be used instead. 

For `pixi exec` its a bit tricky because you might actually want to run on `osx-arm64` instead of `osx-64`. `pixi exec` does already support `--platform` though.

**Comment by AntoinePrv** [comment]
@baszalmstra I would ideally like to use both in the same workspace by simply changing a flag.
That way I could test code locally with both `ARM64` and `x86-64`.

**Comment by baszalmstra** [comment]
Right, so essentially overwrite the default detection and use `pixi run --platform ...` ? I can see that being useful indeed!

**Comment by AntoinePrv** [comment]
Exactly, since this involves having two set of platforms under `.pixi`, let me mention a related use case (unrelated to OP so I don't want to hijack this conversation too much).

Similarly, when I do some C++, I may be interested to test locally on Linux. In this case I get some image with Docker Desktop (it can run Linux `x86-64` and `arm64` on MacOS) and bind-mount my working directory in the Docker image. Now the `.pixi` folders of MacOS and Linux are conflicting.
Being able to tell them apart would help (detached environments may be another direction).
